ibis/sql/postgres/client.py

```diff
@@ -65,19 +71,16 @@ def __init__(self, host=None, user=None, password=None, port=None,
         self.name = url.database
         self.database_name = 'public'
         self.con = sa.create_engine(url)
+        self.inspector = Inspector.from_engine(self.con)
         self.meta = sa.MetaData(bind=self.con)
 
     @property
     def current_database(self):
         return self.database_name
 
     def list_databases(self):
-        # http://dba.stackexchange.com/a/1304/58517
-        return [
-            row.datname for row in self.con.execute(
-                'SELECT datname FROM pg_database WHERE NOT datistemplate'
-            )
-        ]
+        """list databases here means list schemas"""
+        return self.inspector.get_schema_names()
```

Comment:
In databases like SQLite where there's no notion of multiple schemas in the same `.db` file, we can raise an error informing the user that what they are trying to do can't be made to make sense.